src/modules/main/templates/hooks/useFilters.ts

```diff
@@ -30,6 +29,9 @@ const parseDictionary = (key: 'patternsIds' | 'technologiesIds') => (value: stri
     ? filters
     : { ...filters, [key]: value };
 
+const pipe = (...fns: Function[]) => (filters: TemplatesSearchFilters) =>
+  fns.reduce((v, f) => f(v), filters);
+
```

Comment:
After some tests and code check i think we can still improve this a little bit. 
Instead of implementing own pipe function which increases code complexity we can can all this parseSomething functions to return single value not whole object and inside useMemo we can do something like this:

page: getPage,
limit: getLimit 
and same for rest

What you think about that ?

Reply:
Agree, sounds like simpler and easier to understand solution. I'll try to implement it.